src/test/python/test_directional_harmonic.py

```diff
@@ -0,0 +1,62 @@
+from functools import partial
+
+import numpy as np
+from pytest import approx, fixture, mark
+
+from pys2let import (
+    analysis_lm2lmn,
+    analysis_adjoint_lmn2lm,
+    synthesis_lmn2lm,
+    synthesis_adjoint_lm2lmn
+)
+
+
+@fixture
+def rng(request):
+    return np.random.default_rng(getattr(request.config.option, "randomly_seed", None))
+
+
+def random_lms(rng, L):
+    return rng.uniform(size=(L * L, 2)) @ [1, 1j]
+
+
+def random_wavlet_lms(rng, L, nwvlts):
+    lms = [random_lms(rng, L) for _ in range(nwvlts + 1)]
+    return lms[0], np.concatenate(lms[1:])
+
+
+@mark.parametrize(
+    "px2wav,wav2px",
+    [
+        (
+            partial(analysis_lm2lmn, spin=0, upsample=1, N=1),
+            partial(analysis_adjoint_lmn2lm, spin=0, upsample=1, N=1),
+        ),
+        (
+            partial(analysis_lm2lmn, spin=2, upsample=1, N=1),
+            partial(analysis_adjoint_lmn2lm, spin=2, upsample=1, N=1),
+        ),
+        (
+            partial(synthesis_adjoint_lm2lmn, spin=0, upsample=1, N=1),
+            partial(synthesis_lmn2lm, spin=0, upsample=1, N=1),
+        ),
+        (
+            partial(synthesis_adjoint_lm2lmn, spin=2, upsample=1, N=1),
+            partial(synthesis_lmn2lm, spin=2, upsample=1, N=1),
+        ),
+    ],
+)
```

Comment:
noice, btw if you're not aware the library [hypothesis](https://hypothesis.readthedocs.io/en/latest/) is great for getting a test to run multiple circumstances